Vork should be released with CrossVersion.full

We transitively depend on scala-compiler via ammonite. I tried out vork in metaconfig where I had `scalaVersion := 2.12.2` and it crashed with the following error

```
[info] Compiling (synthetic)/ammonite/predef/interpBridge.sc
[error] vork.Processor$FileError: /Users/ollie/dev/metaconfig/docs/SUMMARY.md
[info]
[info]
[info]
[info]
[info]
[error] Caused by: java.lang.Exception: Error during predef initialization:
[error] 	at vork.markdown.repl.AmmoniteRepl.$anonfun$new$2(AmmoniteRepl.scala:134)
[error] 	at scala.Option$WithFilter.foreach(Option.scala:257)
[error] 	at vork.markdown.repl.AmmoniteRepl.<init>(AmmoniteRepl.scala:120)
[error] 	at vork.markdown.processors.AmmonitePostProcessor.<init>(AmmonitePostProcessor.scala:12)
[error] 	at vork.markdown.processors.AmmonitePostProcessor$Factory.create(AmmonitePostProcessor.scala:56)
[error] 	at vork.markdown.processors.AmmonitePostProcessor$Factory.create(AmmonitePostProcessor.scala:54)
[error] 	at com.vladsch.flexmark.internal.PostProcessorManager.postProcess(PostProcessorManager.java:71)
[error] 	at com.vladsch.flexmark.internal.PostProcessorManager.processDocument(PostProcessorManager.java:54)
[error] 	at com.vladsch.flexmark.parser.Parser.postProcess(Parser.java:314)
[error] 	at com.vladsch.flexmark.parser.Parser.parse(Parser.java:294)
[error] 	at vork.Processor.handleMarkdown(Processor.scala:33)
[error] 	at vork.Processor.run$1(Processor.scala:53)
[error] 	at vork.Processor.handlePath(Processor.scala:59)
[error] 	at vork.Processor.$anonfun$generateCompleteSite$1(Processor.scala:82)
[error] 	at vork.Processor.$anonfun$generateCompleteSite$1$adapted(Processor.scala:82)
[error] 	at scala.collection.immutable.List.foreach(List.scala:389)
[error] 	at vork.Processor.generateCompleteSite(Processor.scala:82)
[error] 	at vork.Processor.run(Processor.scala:88)
[error] 	at vork.Cli$.main(Cli.scala:16)
[error] 	at vork.Cli.main(Cli.scala)
[error] Caused by: java.lang.AbstractMethodError: Method ammonite/interp/CompilerCompatibility$$anon$2._lockedCount_$eq(I)V is abstract
[error] 	at ammonite.interp.CompilerCompatibility$$anon$2._lockedCount_$eq(CompilerCompatibility.scala)
[error] 	at scala.tools.nsc.typechecker.Namers.$init$(Namers.scala:26)
[error] 	at ammonite.interp.CompilerCompatibility$$anon$2.<init>(CompilerCompatibility.scala:13)
[error] 	at ammonite.interp.CompilerCompatibility$.analyzer(CompilerCompatibility.scala:13)
[error] 	at ammonite.interp.GlobalInitCompat$$anon$2.analyzer$lzycompute(GlobalInitCompat.scala:50)
[error] 	at ammonite.interp.GlobalInitCompat$$anon$2.analyzer(GlobalInitCompat.scala:50)
[error] 	at scala.tools.nsc.Global.computeInternalPhases(Global.scala:629)
[error] 	at scala.tools.nsc.Global.computePhaseDescriptors(Global.scala:675)
[error] 	at scala.tools.nsc.Global.phaseDescriptors$lzycompute(Global.scala:682)
[error] 	at scala.tools.nsc.Global.phaseDescriptors(Global.scala:682)
[error] 	at scala.tools.nsc.Global$Run.<init>(Global.scala:1171)
[error] 	at ammonite.interp.Compiler$$anon$2.<init>(Compiler.scala:252)
[error] 	at ammonite.interp.Compiler$.apply(Compiler.scala:151)
[error] 	at ammonite.interp.CompilerLifecycleManager.init(CompilerLifecycleManager.scala:80)
[error] 	at ammonite.interp.CompilerLifecycleManager.preprocess(CompilerLifecycleManager.scala:49)
[error] 	at ammonite.interp.Interpreter.compileRunBlock$1(Interpreter.scala:466)
[error] 	at ammonite.interp.Interpreter.$anonfun$processAllScriptBlocks$15(Interpreter.scala:523)
[error] 	at ammonite.util.Res$Success.flatMap(Res.scala:61)
[error] 	at ammonite.interp.Interpreter.$anonfun$processAllScriptBlocks$14(Interpreter.scala:522)
[error] 	at ammonite.util.Res$Success.flatMap(Res.scala:61)
[error] 	at ammonite.interp.Interpreter.$anonfun$processAllScriptBlocks$12(Interpreter.scala:519)
[error] 	at scala.Option.getOrElse(Option.scala:121)
[error] 	at ammonite.interp.Interpreter.loop$1(Interpreter.scala:519)
[info] /Users/ollie/dev/metaconfig/vork/target/vork
[error] 	at ammonite.interp.Interpreter.processAllScriptBlocks(Interpreter.scala:555)
[error] 	at ammonite.interp.Interpreter.$anonfun$processModule$6(Interpreter.scala:354)
[error] 	at ammonite.util.Catching.flatMap(Res.scala:114)
[error] 	at ammonite.interp.Interpreter.$anonfun$processModule$5(Interpreter.scala:345)
[error] 	at ammonite.util.Res$Success.flatMap(Res.scala:61)
[error] 	at ammonite.interp.Interpreter.processModule(Interpreter.scala:337)
[error] 	at ammonite.interp.Interpreter.$anonfun$initializePredef$1(Interpreter.scala:109)
[error] 	at ammonite.interp.Interpreter.$anonfun$initializePredef$1$adapted(Interpreter.scala:109)
[error] 	at ammonite.interp.PredefInitialization$.$anonfun$apply$7(PredefInitialization.scala:65)
[error] 	at ammonite.util.Res$.$anonfun$fold$1(Res.scala:31)
[error] 	at scala.collection.LinearSeqOptimized.foldLeft(LinearSeqOptimized.scala:122)
[error] 	at scala.collection.LinearSeqOptimized.foldLeft$(LinearSeqOptimized.scala:118)
[error] 	at scala.collection.immutable.List.foldLeft(List.scala:86)
[error] 	at ammonite.util.Res$.fold(Res.scala:29)
[error] 	at ammonite.interp.PredefInitialization$.apply(PredefInitialization.scala:53)
[error] 	at ammonite.interp.Interpreter.initializePredef(Interpreter.scala:111)
[error] 	... 18 more
```

Upgrading metaconfig to 2.12.4 fixed the problem.

--- a/vork/src/main/scala/vork/markdown/vorkParserExtension.scala
+++ b/vork/src/main/scala/vork/markdown/vorkParserExtension.scala
@@ -23,17 +23,26 @@ class VorkParserExtension(options: Options) extends Parser.ParserExtension {
       * Creating a custom node extending `ContentNode` or `CustomNode` did not help either.
       */
     override def createNode(nodeChars: BasedSequence): ast.Node = {
-      val key = nodeChars.midSequence(2, nodeChars.length() - 1).trim()
-      val value = site.getOrElse(key.unescape(), sys.error(s"Missing '$key' site variable."))
-      new ast.Text(PrefixedSubSequence.of(value, nodeChars).removeSuffix(nodeChars))
+      nodeChars.toString match {
+        case VariableInjectionPattern(key) =>
+          val value = site.getOrElse(key, sys.error(s"Missing '$key' site variable."))
+          new ast.Text(PrefixedSubSequence.of(value, nodeChars).removeSuffix(nodeChars))
+        case _ =>
+          sys.error("Flexmark matched a variable injection which is not of the expected shape.")
+      }
     }
 
+    private final val totalLength = document.getTextLength
+    private final val VariableInjectionPattern = "!\\[([^\\]\\[]*)\\]".r
     override def isMatch(nodeChars: BasedSequence): Boolean = {
-      // It has to be of the shape `![key]` and not `![key]()` or `![](key)` or `![][key]`
-      nodeChars.countChars(nodeChars) > 3 &&
-      nodeChars.startsWith(CharSubSequence.of("![")) &&
-      nodeChars.endsWith(CharSubSequence.of("]")) &&
-      !nodeChars.containsAllOf(CharSubSequence.of("]["))
+      val matches = nodeChars.toString.matches(VariableInjectionPattern.regex)
+      val startNext = nodeChars.getEndOffset
+      val endNext = startNext + 1
+      // As bracket nesting level is 0, we need to check that the following char does not start with `[`
+      matches && (
+        startNext == totalLength ||
+        !document.getChars.baseSubSequence(startNext, endNext).startsWith("[")
+      )
     }
 
     override def adjustInlineText(doc: ast.Document, node: ast.Node): BasedSequence = node.getChars

--- a/vork/src/test/scala/vork/markdown/processors/SiteVariableInjectorSuite.scala
+++ b/vork/src/test/scala/vork/markdown/processors/SiteVariableInjectorSuite.scala
@@ -36,10 +36,9 @@ class SiteVariableInjectorSuite extends BaseMarkdownSuite {
     """.stripMargin
   )
 
-  // We're missing one leading `!` in the expected, issue has been reported upstream
   check(
     "travis",
-    "# Title ![travis][travis-image]",
+    "# Title [travis][travis-image]",
     "# Title [travis][travis-image]"
   )
 

--- a/vork/src/test/scala/vork/markdown/processors/SiteVariableInjectorSuite.scala
+++ b/vork/src/test/scala/vork/markdown/processors/SiteVariableInjectorSuite.scala
@@ -35,4 +35,11 @@ class SiteVariableInjectorSuite extends BaseMarkdownSuite {
       || 1.0 | hello |
     """.stripMargin
   )
+
+  // We're missing one leading `!` in the expected, issue has been reported upstream
+  check(
+    "travis",
+    "# Title ![travis][travis-image]",
+    "# Title [travis][travis-image]"
+  )
 }

--- a/vork/src/test/scala/vork/markdown/processors/SiteVariableInjectorSuite.scala
+++ b/vork/src/test/scala/vork/markdown/processors/SiteVariableInjectorSuite.scala
@@ -42,4 +42,30 @@ class SiteVariableInjectorSuite extends BaseMarkdownSuite {
     "# Title ![travis][travis-image]",
     "# Title [travis][travis-image]"
   )
+
+  check(
+    "markdown-safe-buffer",
+    """# safe-buffer [![travis][travis-image]][travis-url] [![npm][npm-image]][npm-url] [![downloads][downloads-image]][downloads-url] [![javascript style guide][standard-image]][standard-url]
+      |
+      |[travis-image]: https://img.shields.io/travis/feross/safe-buffer/master.svg
+      |[travis-url]: https://travis-ci.org/feross/safe-buffer
+      |[npm-image]: https://img.shields.io/npm/v/safe-buffer.svg
+      |[npm-url]: https://npmjs.org/package/safe-buffer
+      |[downloads-image]: https://img.shields.io/npm/dm/safe-buffer.svg
+      |[downloads-url]: https://npmjs.org/package/safe-buffer
+      |[standard-image]: https://img.shields.io/badge/code_style-standard-brightgreen.svg
+      |[standard-url]: https://standardjs.com
+    """.stripMargin,
+    """# safe-buffer [![travis][travis-image]][travis-url] [![npm][npm-image]][npm-url] [![downloads][downloads-image]][downloads-url] [![javascript style guide][standard-image]][standard-url]
+      |
+      |[travis-image]: https://img.shields.io/travis/feross/safe-buffer/master.svg
+      |[travis-url]: https://travis-ci.org/feross/safe-buffer
+      |[npm-image]: https://img.shields.io/npm/v/safe-buffer.svg
+      |[npm-url]: https://npmjs.org/package/safe-buffer
+      |[downloads-image]: https://img.shields.io/npm/dm/safe-buffer.svg
+      |[downloads-url]: https://npmjs.org/package/safe-buffer
+      |[standard-image]: https://img.shields.io/badge/code_style-standard-brightgreen.svg
+      |[standard-url]: https://standardjs.com
+    """.stripMargin
+  )
 }

